src/main/haskell/kore/src/Kore/Logger.hs

```diff
@@ -0,0 +1,128 @@
+{-|
+Module      : Kore.Logger
+Description : Logging functions.
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+Maintainer  : vladimir.ciobanu@runtimeverification.com
+Stability   : experimental
+Portability : portable
+-}
+
+module Kore.Logger
+    ( LogMessage (..)
+    , Severity (..)
+    , log
+    , logDebug
+    , logInfo
+    , logWarning
+    , logError
+    , logCritical
+    , withLogScope
+    ) where
+
+import           Colog
+                 ( WithLog )
+import qualified Colog as Colog
+import           Data.Functor.Contravariant
+                 ( contramap )
+import           Data.Text
+                 ( Text )
+import qualified Data.Text as Text
+import           Prelude hiding
+                 ( log )
+
+-- | Log level used to describe each log message. It is also used to set the
+-- minimum level to be outputted.
+data Severity
+    = Debug
+    -- ^ Lowest level used for low-level debugging.
+    | Info
+    -- ^ Used for various informative messages.
+    | Warning
+    -- ^ Used for odd/unusual cases which are recoverable.
+    | Error
+    -- ^ Used for application errors, unexpected behaviors, etc.
+    | Critical
+    -- ^ Used before shutting down the application.
+    deriving (Show, Read, Eq, Ord)
+
+-- | This type should not be used directly, but rather should be created and
+-- dispatched through the `log` functions.
+data LogMessage = LogMessage
```

Comment:
The field names here do not conform to the [style guide](https://github.com/kframework/kore/wiki/Coding-Standards#records).